packages/template-retail-react-app/app/pages/product-list/partials/radio-refinements.jsx

```diff
@@ -12,7 +12,11 @@ import PropTypes from 'prop-types'
 const RadioRefinements = ({filter, toggleFilter, selectedFilters}) => {
     return (
         <Box>
-            <RadioGroup value={selectedFilters}>
+            <RadioGroup
+                // The following `false` fallback is required to avoid the radio group
+                // from switching to "uncontrolled mode" when `selectedFilters` is empty.
+                value={selectedFilters[0] ?? false}
```

Comment:
This is the key to fixing item 6 from the points listed in the description.

If `RadioGroup` receives undefined or null in the `value` prop, it no longer acts in controlled mode and doesn't handle the checked the state of inner radio buttons.

This _could_ cause issues if a refinement had an option where `.value` is equal to `false`. In that case the `false` fallback would just need to be replaced with any other non-null value (e.g. `''`).

Reply:
Nice catch.